packages/migrate/utils.js

```diff
@@ -205,54 +205,65 @@ export function posixify(str) {
  * @param {Array<[string, string, string?, ('dependencies' | 'devDependencies')?]>} updates
  */
 export function update_pkg(content, updates) {
-	const indent = content.split('\n')[1].match(/^\s+/)?.[0] || '  ';
-	const pkg = JSON.parse(content);
-
-	/**
-	 * @param {string} name
-	 * @param {string} version
-	 * @param {string} [additional]
-	 * @param {'dependencies' | 'devDependencies' | undefined} [insert]
-	 */
-	function update_pkg(name, version, additional = '', insert) {
-		if (pkg.dependencies?.[name]) {
-			const existing_range = pkg.dependencies[name];
-
-			if (semver.validRange(existing_range) && !semver.subset(existing_range, version)) {
-				log_migration(`Updated ${name} to ${version} ${additional}`);
-				pkg.dependencies[name] = version;
-			}
-		}
-
-		if (pkg.devDependencies?.[name]) {
-			const existing_range = pkg.devDependencies[name];
-
-			if (semver.validRange(existing_range) && !semver.subset(existing_range, version)) {
-				log_migration(`Updated ${name} to ${version} ${additional}`);
-				pkg.devDependencies[name] = version;
-			}
-		}
-
-		if (insert && !pkg[insert]?.[name]) {
-			if (!pkg[insert]) pkg[insert] = {};
-
-			// Insert the property in sorted position without adjusting other positions so diffs are easier to read
-			const sorted_keys = Object.keys(pkg[insert]).sort();
-			const index = sorted_keys.findIndex((key) => name.localeCompare(key) === -1);
-			const insert_index = index !== -1 ? index : sorted_keys.length;
-			const new_properties = Object.entries(pkg[insert]);
-			new_properties.splice(insert_index, 0, [name, version]);
-			pkg[insert] = Object.fromEntries(new_properties);
-
-			log_migration(`Added ${name} version ${version} ${additional}`);
-		}
-	}
-
-	for (const update of updates) {
-		update_pkg(...update);
-	}
-
-	return JSON.stringify(pkg, null, indent);
+    const indent = content.split('\n')[1].match(/^\s+/)?.[0] || '  ';
```

Comment:
formatting looks wrong (spaces instead of tabs). I just realized that the `migrate` package doesn't have the related scripts: Could you add
```json
		"lint": "prettier --check .",
		"format": "pnpm lint --write",
```
to the `package.json`? Then you can format with `pnpm format` and CI would now fail correctly with wrong formatting

Reply:
should the `pnpm format` be separated PR ? because the change is a lot

Reply:
maybe, I will add only related file for this PR

Reply:
From my local testing it only changes a couple of files, and only in a few places, I'm ok with that.